phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIndexIdRetrieveIT.java

```diff
@@ -0,0 +1,169 @@
+import java.util.Map;
+import java.util.Properties;
+
+import com.google.common.collect.Maps;
+import org.apache.hadoop.hbase.regionserver.ScanInfoUtil;
+import org.apache.phoenix.util.PropertiesUtil;
+import org.apache.phoenix.util.ReadOnlyProps;
+import org.apache.phoenix.util.SchemaUtil;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ViewIndexIdRetrieveIT extends ParallelStatsDisabledIT {
+    private String ddlForBaseTable = "CREATE TABLE %s (TENANT_ID CHAR(15) NOT NULL, ID CHAR(3)" +
+            " NOT NULL, NUM BIGINT CONSTRAINT PK PRIMARY KEY (TENANT_ID, ID))" +
+            " MULTI_TENANT = true, COLUMN_ENCODED_BYTES=0 ";
+    private String ddlForView = "CREATE VIEW %s (A BIGINT PRIMARY KEY, B BIGINT)" +
+            " AS SELECT * FROM %s WHERE ID='ABC'";
+    private String ddlForViewIndex =
+            "CREATE INDEX %s ON %s (B DESC) INCLUDE (NUM)";
+    private String selectAll = "SELECT * FROM SYSTEM.CATALOG";
+    private String selectRow = "SELECT VIEW_INDEX_ID,VIEW_INDEX_ID_DATA_TYPE" +
+            " FROM SYSTEM.CATALOG WHERE TABLE_NAME='%s' AND COLUMN_COUNT IS NOT NULL";
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+        props.put(ScanInfoUtil.PHOENIX_MAX_LOOKBACK_AGE_CONF_KEY, Integer.toString(60*60));
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+    }
+
+    @Test
+    public void testSelectViewIndexIdAsLong() throws Exception {
+        testSelectViewIndexId(true);
+    }
+
+    @Test
+    public void testSelectViewIndexIdAsShort() throws Exception {
+        testSelectViewIndexId(false);
+    }
+
+    private void testSelectViewIndexId(boolean isTestingLongViewIndexId) throws Exception {
+        String val = isTestingLongViewIndexId ? "true" : "false";
+        int expectedDataType = isTestingLongViewIndexId ? Types.BIGINT : Types.SMALLINT;
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        props.setProperty(LONG_VIEW_INDEX_ENABLED_ATTRIB, val);
+        String schema = generateUniqueName();
+        String baseTable = generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(schema, baseTable);
+        String viewName = generateUniqueName();
+        String viewFullName = SchemaUtil.getTableName(schema, viewName);
+        String viewIndexName = generateUniqueName();
+        try (final Connection conn = DriverManager.getConnection(url,props);
+             final Statement stmt = conn.createStatement()) {
+            stmt.execute(String.format(ddlForBaseTable, fullTableName));
+            stmt.execute(String.format(ddlForView, viewFullName, fullTableName));
+            stmt.execute(String.format(ddlForViewIndex, viewIndexName, viewFullName));
+
+            ResultSet rs = stmt.executeQuery(String.format(selectRow, viewIndexName));
+            rs.next();
+            assertEquals(Short.MIN_VALUE,rs.getLong(1));
```

Comment:
Even when the `LONG_VIEW_INDEX_ENABLED_ATTRIB ` config is set to long, we expect Short.MIN_VALUE to allow clients to switch the config (as we discussed offline). Can you add a comment about this in the test?